### PR TITLE
Add doxygen build to cmake

### PR DESCRIPTION
**Pull request by Acuadros95**

### CODE REVIEW

File: include/uxr/client/core/session/create_entities_xml.h [hunk 1]
```diff
@@ -30,7 +34,7 @@ extern "C"
  * @param session       A uxrSession structure previously initialized.
  * @param stream_id     The output stream identifier where the CREATE submessage will be buffered.
  * @param object_id     The identifier of the XRCE Participant.
- * @param domain_id     The identifier of the Domain to which the XRCE Participant belongs.
+ * @param domain        The identifier of the Domain to which the XRCE Participant belongs.
```
Comment (pablogs9) [inline include/uxr/client/core/session/create_entities_xml.h:37]: why not `domain_id`?

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Linux/331/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Windows/344/)

File: include/uxr/client/core/session/create_entities_xml.h (hunk 1 above)
Comment (Acuadros95) [reply include/uxr/client/core/session/create_entities_xml.h:37]: Because the @param name must match the function parameter name, which is `domain`:

https://github.com/eProsima/Micro-XRCE-DDS-Client/blob/7c3e2f491191370e158809736cc3842b2754bb13/include/uxr/client/core/session/create_entities_xml.h#L45-L50

Comment (pablogs9) [reply include/uxr/client/core/session/create_entities_xml.h:37]: Let's change all to `domain_id`

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Linux/336/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Windows/349/)

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Linux/339/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Windows/352/)

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Linux/340/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Windows/353/)

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Linux/342/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Windows/355/)

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -29,7 +29,7 @@ Contributors must sign-off each commit by adding a `Signed-off-by: ...` line to
 
 ## Code Coverage
 
-As stated in [QUALITY.md](QUALITY.md), all contributions to the project must increase line code coverage.
+As stated in [QUALITY.md](https://github.com/eProsima/Micro-XRCE-DDS-Client/blob/master/QUALITY.md), all contributions to the project must increase line code coverage.
```
Comment (pablogs9) [inline CONTRIBUTING.md:32]:
```suggestion
As stated in [QUALITY.md](QUALITY.md), all contributions to the project must increase line code coverage.
```

File: QUALITY.md [hunk 3]
```diff
@@ -12,7 +12,7 @@ Below are the rationales, notes and caveats for this claim, organized by each re
 
 ### Version Scheme [1.i]
 
-The **Versioning Policy Declaration** for **eProsima Micro XRCE-DDS Client** can be found [here](VERSIONING.md) and it adheres to [`semver`](https://semver.org/).
+The **Versioning Policy Declaration** for **eProsima Micro XRCE-DDS Client** can be found [here](https://github.com/eProsima/Micro-XRCE-DDS-Client/blob/master/VERSIONING.md) and it adheres to [`semver`](https://semver.org/).
```
Comment (pablogs9) [inline QUALITY.md:15]:
```suggestion
The **Versioning Policy Declaration** for **eProsima Micro XRCE-DDS Client** can be found [here](VERSIONING.md) and it adheres to [`semver`](https://semver.org/).
```

File: QUALITY.md [hunk 4]
```diff
@@ -34,7 +34,7 @@ Any ABI break in **eProsima Micro XRCE-DDS Client** will be done between minor v
 ## Change Control Process [2]
 
 The stability of **eProsima Micro XRCE-DDS Client** is ensured through reviews, CI and tests.
-The change control process can be found in [CONTRIBUTING](CONTRIBUTING.md)
+The change control process can be found in [CONTRIBUTING](https://github.com/eProsima/Micro-XRCE-DDS-Client/blob/master/CONTRIBUTING.md)
```
Comment (pablogs9) [inline QUALITY.md:37]:
```suggestion
The change control process can be found in [CONTRIBUTING](CONTRIBUTING.md)
```

File: QUALITY.md [hunk 5]
```diff
@@ -47,7 +47,7 @@ All changes will occur through a pull request.
 ### Contributor Origin [2.ii]
 
 **eProsima Micro XRCE-DDS Client** uses the [Developer Certificate of Origin (DCO)](https://developercertificate.org/) as its confirmation of contributor origin policy since version 2.0.0.
-More information can be found in [CONTRIBUTING](CONTRIBUTING.md)
+More information can be found in [CONTRIBUTING](https://github.com/eProsima/Micro-XRCE-DDS-Client/blob/master/CONTRIBUTING.md)
```
Comment (pablogs9) [inline QUALITY.md:50]:
```suggestion
More information can be found in [CONTRIBUTING](CONTRIBUTING.md)
```

File: QUALITY.md [hunk 6]
```diff
@@ -64,7 +64,7 @@ Current nightly results for all supported platforms can be checked at the links:
 
 ### Documentation Policy [2.v]
 
-All pull requests must resolve related documentation changes before merging as stated in [CONTRIBUTING](CONTRIBUTING.md).
+All pull requests must resolve related documentation changes before merging as stated in [CONTRIBUTING](https://github.com/eProsima/Micro-XRCE-DDS-Client/blob/master/CONTRIBUTING.md).
```
Comment (pablogs9) [inline QUALITY.md:67]:
```suggestion
All pull requests must resolve related documentation changes before merging as stated in [CONTRIBUTING](CONTRIBUTING.md).```

File: QUALITY.md [hunk 7]
```diff
@@ -90,7 +90,7 @@ The **eProsima Micro XRCE-DDS Client** copyright holder provides a statement of
 ### Feature Testing [4.i]
 
 **eProsima Micro XRCE-DDS Client** provides tests which simulate typical usage, and they are located in the [`test` directory](test).
-New features are required to have tests before being added as stated in [CONTRIBUTING](CONTRIBUTING.md).
+New features are required to have tests before being added as stated in [CONTRIBUTING](https://github.com/eProsima/Micro-XRCE-DDS-Client/blob/master/CONTRIBUTING.md).
```
Comment (pablogs9) [inline QUALITY.md:93]:
```suggestion
New features are required to have tests before being added as stated in [CONTRIBUTING](CONTRIBUTING.md).
```

File: QUALITY.md [hunk 8]
```diff
@@ -112,7 +112,7 @@ The tests aim to cover typical usage and corner cases.
 3. If the CI system reports a coverage regression after a pull request has been merged, the maintainers must study the case and decide how to proceed, most often reverting the changes and asking for a more thorough testing of the committed changes.
 4. This policy is enforced through the [nightly Micro XRCE-DDS Client CI job](http://jenkins.eprosima.com:8080/view/Micro%20XRCE/job/Micro-XRCE-DDS-Client%20Nightly%20Master%20Linux/).
 
-As stated in [CONTRIBUTING.md](CONTRIBUTING.md), developers and contributors are required to run a line coverage assessment locally before submitting a PR.
+As stated in [CONTRIBUTING.md](https://github.com/eProsima/Micro-XRCE-DDS-Client/blob/master/CONTRIBUTING.md), developers and contributors are required to run a line coverage assessment locally before submitting a PR.
```
Comment (pablogs9) [inline QUALITY.md:115]:
```suggestion
As stated in [CONTRIBUTING.md](CONTRIBUTING.md), developers and contributors are required to run a line coverage assessment locally before submitting a PR.
```

File: QUALITY.md [hunk 9]
```diff
@@ -150,7 +150,7 @@ The latest pull request results can be seen [here](http://jenkins.eprosima.com:8
 * Linux [![Linux ci](http://jenkins.eprosima.com:8080/view/Nightly/job/Micro-XRCE-DDS-Client%20Nightly%20Master%20Linux/badge/icon?subject=%20%20%20Linux%20CI%20)](http://jenkins.eprosima.com:8080/view/Micro%20XRCE/job/Micro-XRCE-DDS-Client%20Nightly%20Master%20Linux/)
 * Windows [![Windows ci](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Nightly%20Master%20Windows/badge/icon?subject=%20%20%20%20Windows%20CI%20)](http://jenkins.eprosima.com:8080/view/Micro%20XRCE/job/Micro-XRCE-DDS-Client%20Nightly%20Master%20Windows/)
 
-More information about the supported platforms can be found in [PLATFORM_SUPPORT](PLATFORM_SUPPORT.md)
+More information about the supported platforms can be found in [PLATFORM_SUPPORT](https://github.com/eProsima/Micro-XRCE-DDS-Client/blob/master/PLATFORM_SUPPORT.md)
```
Comment (pablogs9) [inline QUALITY.md:153]:
```suggestion
More information about the supported platforms can be found in [PLATFORM_SUPPORT](PLATFORM_SUPPORT.md)
```

File: README.md [hunk 10]
```diff
@@ -36,7 +36,7 @@ You can access the *eProsima Micro XRCE-DDS* user documentation online, which is
 ## Quality Declaration
 
 **eProsima Micro XRCE-DDS Client** claims to be in the **Quality Level 1** category based on the guidelines provided by [ROS 2](https://ros.org/reps/rep-2004.html).
-See the [Quality Declaration](QUALITY.md) for more details.
+See the [Quality Declaration](https://github.com/eProsima/Micro-XRCE-DDS-Client/blob/master/QUALITY.md) for more details.
```
Comment (pablogs9) [inline README.md:39]:
```suggestion
See the [Quality Declaration](QUALITY.md) for more details.
```

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Linux/344/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Windows/357/)

**Review by pablogs9 — APPROVED** [review]
